tests/Mawosoft.PowerShell.WindowsSearchManager.Tests/UpdateSearchCatalogCommandTests.cs

```diff
@@ -0,0 +1,86 @@
+    [InlineData(@"Update-SearchCatalog -All", "Reindex()", "")]
+    [InlineData(@"Update-SearchCatalog -RootPath x:\", "ReindexSearchRoot({0})", @"x:\")]
+    [InlineData(@"Update-SearchCatalog -RootPath x:\, y:\", "ReindexSearchRoot({0})", @"x:\", @"y:\")]
+    [InlineData(@"@([pscustomobject]@{ RootPath = 'x:\'}, [pscustomobject]@{ RootPath = 'y:\'}) | Update-SearchCatalog", "ReindexSearchRoot({0})", @"x:\", @"y:\")]
+    [InlineData(@"Update-SearchCatalog -Path x:\foo", "ReindexMatchingURLs({0})", @"x:\foo")]
+    [InlineData(@"Update-SearchCatalog -Path x:\foo, x:\bar", "ReindexMatchingURLs({0})", @"x:\foo", @"x:\bar")]
+    [InlineData(@"Update-SearchCatalog x:\foo, x:\bar", "ReindexMatchingURLs({0})", @"x:\foo", @"x:\bar")]
+    [InlineData(@"@('x:\foo', 'x:\bar') | Update-SearchCatalog", "ReindexMatchingURLs({0})", @"x:\foo", @"x:\bar")]
+    [InlineData(@"@([pscustomobject]@{ Path = 'x:\foo'}, [pscustomobject]@{ Path = 'x:\bar'}) | Update-SearchCatalog", "ReindexMatchingURLs({0})", @"x:\foo", @"x:\bar")]
+    public void Command_Succeeds(string script, string expectedMethodcall, params string[] expectedMethodParameters)
+    {
+        Collection<PSObject> results = InvokeScript(script);
+        Assert.Empty(results);
+        Assert.False(PowerShell.HadErrors);
+        Assert.All(InterfaceChain.CatalogManager.RecordedCalls,
+            (item, i) => Assert.Equal(string.Format(expectedMethodcall, expectedMethodParameters[i]), item));
+        Assert.Equal(expectedMethodParameters.Length, InterfaceChain.CatalogManager.RecordedCalls.Count);
+        Assert.True(InterfaceChain.SingleHasWriteRecordings(InterfaceChain.CatalogManager));
+    }
+
+    [Fact]
+    public void AllParameterSet_NoAdmin_FailsWithCustomMessage()
+    {
+        InterfaceChain.CatalogManager.AdminMode = false;
+        Collection<PSObject> results = InvokeScript("Update-SearchCatalog -All ");
+        Assert.Empty(results);
+        AssertUnauthorizedAccess();
+    }
+
+    [Theory]
+    [ClassData(typeof(Exception_TheoryData))]
+    public void AllParameterSet_HandlesFailures(ExceptionParam exceptionParam)
+    {
+        InterfaceChain.CatalogManager.AddException("^Reindex$", exceptionParam.Exception);
+        Collection<PSObject> results = InvokeScript("Update-SearchCatalog -All ");
+        Assert.Empty(results);
+        AssertSingleErrorRecord(exceptionParam);
+    }
+
+    [Theory]
+    [InlineData("RootPath", true)]
+    [InlineData("RootPath", false)]
+    [InlineData("Path", true)]
+    [InlineData("Path", false)]
+    public void RootAndPathParameterSet_WithFailures_PartiallySucceeds(string parameterName, bool usePipeline)
+    {
+        string expectedMethodcall = parameterName == "Path" ? "ReindexMatchingURLs({0})" : "ReindexSearchRoot({0})";
+        string exceptionRegex = parameterName == "Path" ? "" : "";
+
+        InterfaceChain.CatalogManager.AddException("^ReindexSearchRoot$|^ReindexMatchingURLs$", new Exception(), 2, 4);
```

Comment:
Use local variable `exceptionRegex` instead of literal. Or keep the literal and remove the variable.